### DISCUSSION TITLE
Associating inventory item with component

### DISCUSSION THREAD

**Discussion by candlerb** [netbox-community/netbox#11085] [Q&A] [answered]
I may be being dense here.

I know it was possible in the past to install an Inventory Item in a device, and associate it with a particular Component.  (For example, an SFP module inserted into a specific Interface).

There is still a "Component" column available in the list of Inventory Items on a device, e.g. `/dcim/devices/1/inventory/`

However with 3.3.9, when I add a new inventory item or edit an existing inventory item, I cannot see how to associate it with a particular component.  What am I missing?

**Comment by kkthxbye-code** [comment, marked as answer]
> However with 3.3.9, when I add a new inventory item or edit an existing inventory item, I cannot see how to associate it with a particular component. What am I missing?

The only way to associate an inventory item with a component through the UI is to go to the component detail view and press "add inventory item". The most logical way would be to edit the inventory item and assign the related component. I didn't implement the association, but my guess as to why it wasn't done is that the relationship is a generic foreign key, which makes UI selection hard as you would have to do something like this:

![image](https://user-images.githubusercontent.com/400797/205263161-0d3f11ff-3a79-4833-b8ea-38bc8a75f564.png)

But for all component types. We can't do a single dropdown as there's no single endpoint for getting device components. I do think we need a solution though, so feel free to create a FR. Maybe @jeremystretch knows a magic solution.

**Reply by candlerb** [reply]
Ah yes, thank you.  In fact, I had gone to the interface *edit* page to try to find the relationship in the opposite direction, and it wasn't there.  But I didn't try the interface *view details* page.  This has caught me out in the past with FHRP groups too.

What I was actually trying to find out is whether it's possible to assign to any type of component: e.g. can an inventory item be assigned to a power port?  The answer is yes: I can go to the power port view details page, and add an inventory item.  Therefore, as you say, it would have to be generic "Component Assignment".

I think the main benefit of having this assignment available, apart from making it easier to find, would be that it would be possible to reassign an inventory item to another component.  But if you're going to allow that, I think you'd also want to be able to move an inventory item to a different device.  I don't think that's possible today, even if it's not assigned to a component: at `/dcim/inventory-items/1/edit/` the device selector is greyed out.  So in that case, you have to delete it and recreate it.

Hence I'm not really sure what the scope of the FR should be.
